chainer/variable.py

```diff
@@ -1464,15 +1498,43 @@ def __bool__(self):
     __hash__ = None
 
 
+if six.PY3:
+    OrderedDict = collections.OrderedDict
+else:
+    # Reference counting cannot free keys in old `collections.OrderedDict`,
+    # where a doubly linked list is used to maintain the order.
+    class OrderedDict(object):
```

Comment:
How about moving it to `utils`?
Another (more minor) point: can we use a plain `dict` for py>3.6?